src/python_testing/TC_PWRTL_2_1.py

```diff
@@ -40,36 +40,43 @@ async def test_TC_PWRTL_2_1(self):
 
         attributes = Clusters.PowerTopology.Attributes
 
-        endpoint = self.user_params.get("endpoint", 1)
+        endpoint = 1
 
-        self.print_step(1, "Commissioning, already done")
+        powertop_attr_list = Clusters.Objects.PowerTopology.Attributes.AttributeList
+        powertop_cluster = Clusters.Objects.PowerTopology
+        attribute_list = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=powertop_cluster, attribute=powertop_attr_list)
+        avail_endpoints_attr_id = Clusters.Objects.PowerTopology.Attributes.ActiveEndpoints.attribute_id
+        act_endpoints_attr_id = Clusters.Objects.PowerTopology.Attributes.AvailableEndpoints.attribute_id
 
-        if not self.check_pics("PWRTL.S.A0000"):
-            logging.info("Test skipped because PICS PWRTL.S.A0000 is not set")
-            return
+        self.print_step(1, "Commissioning, already done")
 
         self.print_step(2, "Read AvailableAttributes attribute")
-        available_endpoints = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.PowerTopology, attribute=attributes.AvailableEndpoints)
-
-        if available_endpoints == NullValue:
-            logging.info("AvailableEndpoints is null")
-        else:
-            logging.info("AvailableEndpoints: %s" % (available_endpoints))
+        if avail_endpoints_attr_id in attribute_list:
+            available_endpoints = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.PowerTopology, attribute=attributes.AvailableEndpoints)
 
-            asserts.assert_less_equal(len(available_endpoints), 21,
-                                      "AvailableEndpoints length %d must be less than 21!" % len(available_endpoints))
+            if available_endpoints == NullValue or available_endpoints == []:
+                logging.info("AvailableEndpoints is null or an empty list")
+            else:
+                logging.info("AvailableEndpoints: %s" % (available_endpoints))
+                asserts.assert_less_equal(len(available_endpoints), 21,
+                                          "AvailableEndpoints length %d must be less than 21!" % len(available_endpoints))
```

Comment:
I know this isn't your code, or test plan, but doesn't it feel like there's a step missing here? Shouldn't there be a check that the listed points actually exist? Do you know if that check is done elsewhere for this cluster?

Reply:
Hi Cecille,
I also feel like there might be something missing here, but my understanding and knowledge here is limited. As this appears to be the only power topology clusters test currently, I don't believe there is another place that checks if available endpoints exist's. I think the closest thing might be the if check above for if AvailableEndpoints is an empty list. Should an additional check be added here to make sure that endpoints exist?

Reply:
kk, opened an issue here - https://github.com/project-chip/matter-test-scripts/issues/369. I assigned it to you. I will need a minor test plan change. basically just adding a sentence in the expected column saying that all the endpoints are endpoints that exist on the device. And then that can just be added here. Not a high priority, but if you want to knock out a quick one, feel free to have a go at it.

Reply:
feel free to resolve this comment, btw...nothing to do on this PR, we can continue on the issue.